tests/integration/test_diff.py

```diff
@@ -0,0 +1,353 @@
+                        'change_type': ChangeType.REMOVED,
+                    },
+                },
+                {
+                    'subpath': 'fifo1',
+                    'data': {
+                        'file_type': FileType.FIFO,
+                        'change_type': ChangeType.REMOVED,
+                    },
+                },
+                {
+                    'subpath': 'file1',
+                    'data': {
+                        'file_type': FileType.FILE,
+                        'change_type': ChangeType.REMOVED,
+                    },
+                },
+                {
+                    'subpath': 'hardlink1',
+                    'data': {
+                        'file_type': FileType.FILE,
+                        'change_type': ChangeType.REMOVED,
+                    },
+                },
+                {
+                    'subpath': 'symlink1',
+                    'data': {
+                        'file_type': FileType.LINK,
+                        'change_type': ChangeType.REMOVED,
+                    },
+                },
+            ],
+        ),
+        (
+            'test-archive5',
+            'test-archive6',
+            [
+                {
+                    'subpath': 'dir1',
+                    'data': {
+                        'file_type': FileType.FILE,
+                        'change_type': ChangeType.MODIFIED,
+                    },
+                },
+            ],
+        ),
+    ],
+)
+def test_archive_diff_lines(qapp, qtbot, archive_name_1, archive_name_2, expected):
+    params = BorgDiffJob.prepare(vorta.store.models.BackupProfileModel.select().first(), archive_name_1, archive_name_2)
+    thread = BorgDiffJob(params['cmd'], params, qapp)
+
+    with qtbot.waitSignal(thread.result, **pytest._wait_defaults) as blocker:
+        blocker.connect(thread.updated)
+        thread.run()
+
+    diff_lines = blocker.args[0]['data']
+    json_lines = blocker.args[0]['params']['json_lines']
+
+    model = DiffTree()
+    model.setMode(model.DisplayMode.FLAT)
+
+    # Use ParseThread to parse the diff lines
+    parse_thread = ParseThread(diff_lines, json_lines, model)
+    parse_thread.start()
+    qtbot.waitUntil(lambda: parse_thread.isFinished(), **pytest._wait_defaults)
+
+    assert model.rowCount() == len(expected)
+
+    for index, item in enumerate(expected):
+        assert model.index(index, 0).internalPointer().subpath == item['subpath']
+
+        # Checking all attributes for every line will produce very large testing output and will be difficult to debug
+        # So we will check only the attributes we are interested in
+        # TODO: Remove below line above code review
+        # assert model.index(index, 0).internalPointer().data == DiffData(**item['data'])
+
+        for key, value in item['data'].items():
+            assert getattr(model.index(index, 0).internalPointer().data, key) == value
```

Comment:
Old versions of Borg do not support fifo among some other file types (1.1.18) causing this test to fail. See https://github.com/borgbackup/borg/issues/2324

Should I specify this condition explicitly? Perhaps subtracting unsupported file type from length assertion and skipping them in L353 if version is old?

Reply:
Yes, that makes sense.